controllers/kubemon/reconciler_test.go

```diff
@@ -0,0 +1,200 @@
+		fakeClient := fake.NewFakeClientWithScheme(scheme.Scheme,
+			&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{
+				UID:  testUID,
+				Name: kubesystem.Namespace,
+			}},
+			instance)
+		reconciler := NewReconciler(
+			fakeClient, fakeClient, scheme.Scheme, dtcMock, log, &corev1.Secret{}, instance,
+		)
+		connectionInfo := dtclient.ConnectionInfo{TenantUUID: testUID}
+		tenantInfo := &dtclient.TenantInfo{ID: testUID}
+
+		dtcMock.
+			On("GetConnectionInfo").
+			Return(connectionInfo, nil)
+
+		dtcMock.
+			On("GetTenantInfo").
+			Return(tenantInfo, nil)
+
+		assert.NotNil(t, reconciler)
+
+		result, err := reconciler.Reconcile(request)
+
+		assert.NoError(t, err)
+		assert.NotNil(t, result)
+
+		var statefulSet v1.StatefulSet
+		err = fakeClient.Get(context.TODO(), client.ObjectKey{Name: instance.Name + StatefulSetSuffix, Namespace: instance.Namespace}, &statefulSet)
+
+		expected := *newStatefulSet(*instance, testUID)
+		expected.Spec.Template.Spec.Volumes = nil
+
+		assert.NoError(t, err)
+		assert.NotNil(t, statefulSet)
+		assert.Equal(t, expected.ObjectMeta.Name, statefulSet.ObjectMeta.Name)
+		assert.Equal(t, expected.ObjectMeta.Namespace, statefulSet.ObjectMeta.Namespace)
+		assert.Equal(t, expected.Spec, statefulSet.Spec)
+	})
+	t.Run(`Reconcile reconciles custom properties if set`, func(t *testing.T) {
+		log := logf.Log.WithName("TestReconciler")
+		request := reconcile.Request{}
+		dtcMock := &dtclient.MockDynatraceClient{}
+		instance := &v1alpha1.DynaKube{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testName,
+				Namespace: testNamespace,
+			},
+			Spec: v1alpha1.DynaKubeSpec{
+				KubernetesMonitoringSpec: v1alpha1.KubernetesMonitoringSpec{
+					CustomProperties: &v1alpha1.DynaKubeValueSource{
+						Value: testValue,
+					},
+				}}}
+		fakeClient := fake.NewFakeClientWithScheme(scheme.Scheme,
+			&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{
+				UID:  testUID,
+				Name: kubesystem.Namespace,
+			}},
+			instance)
+		reconciler := NewReconciler(
+			fakeClient, fakeClient, scheme.Scheme, dtcMock, log, &corev1.Secret{}, instance,
+		)
+		connectionInfo := dtclient.ConnectionInfo{TenantUUID: testUID}
+		tenantInfo := &dtclient.TenantInfo{ID: testUID}
+
+		dtcMock.
+			On("GetConnectionInfo").
+			Return(connectionInfo, nil)
+
+		dtcMock.
+			On("GetTenantInfo").
+			Return(tenantInfo, nil)
+
+		assert.NotNil(t, reconciler)
+
+		result, err := reconciler.Reconcile(request)
+
+		assert.NoError(t, err)
+		assert.NotNil(t, result)
+
+		var customPropertiesSecret corev1.Secret
+		err = fakeClient.Get(context.TODO(), client.ObjectKey{Name: fmt.Sprintf("%s-%s-%s", instance.Name, Name, customproperties.Suffix), Namespace: testNamespace}, &customPropertiesSecret)
+
+		assert.NoError(t, err)
+		assert.NotNil(t, customPropertiesSecret)
+		assert.Equal(t, testValue, string(customPropertiesSecret.Data[customproperties.DataKey]))
+	})
+	t.Run(`Reconcile posts instance to dashboard if endpoint is set`, func(t *testing.T) {
```

Comment:
Isn't the add to dashboard feature disabled for the time being?

Reply:
afaik it is enabled but not advertised

Reply:
Ah, when KubernetesAPIEndpoint is set. Then, I'm fine with it.